src/js/apps/forms/form/form_app.js

```diff
@@ -283,6 +286,12 @@ export default App.extend({
       'click:save'() {
         Radio.request(`form${ this.form.id }`, 'send', 'form:submit');
         this.showFormSaveDisabled();
+
+        const saveButtonType = this.getState('saveButtonType');
+
+        if (saveButtonType === 'saveAndGoBack') {
+          this.loadingModal = Radio.request('modal', 'show:loading');
```

Comment:
this can't happen in this button because the form can error out prior to even being submitted to the server.

In the form submission inside the form service we are triggering `success` and `error` after the submission to the server.  We need to add a `trigger` there for `submit` where we listen to that event in this app and show the modal.  Then we need to handle the modal for a success or destroy the modal for an error.

Reply:
Ah good call, totally missed this. Updated in the last push.